### PR TITLE
PublicName and dev options fixes

### PR DESCRIPTION
**Pull request by mischievous**
Code to set a public hostname to be used when updating the invitation url by reading from server_config.cfg 
    "general -> publicName".  If the publicName is empty or does not exists it falls back to normal operation.

 Added code to planarally.py to display the warning about the template directory if not running in 
    dev mode.

### CODE REVIEW

**Review by Kruptein — CHANGES_REQUESTED** [review]
In general pretty good stuff, some minor things here and there.

Although this works perfectly as is, I would prefer it if you just provided the public name along with the room info when the session loads ("Room.Info.Set" to be exact).
This simply because the request is launched to the server on load anyway at this moment and would make the logic slightly more easy to reason about. Also there is no reason to expect this to actually change during gameplay as this will require a server restart anyway to fetch the new config value.

File: client/src/game/store.ts [hunk 1]
```diff
@@ -247,6 +249,14 @@ class GameStore extends VuexModule implements GameState {
         this.invitationCode = code;
     }
 
+    @Mutation
+    setPublicName(name: string): void {
+        if (!name.length) return;
+        console.log(name.length);
+        console.log(name);
```
[the remark this answers is not in the document]
Comment (mischievous) [reply client/src/game/store.ts:256]: Removed.

File: server/api/socket/room.py [hunk 2]
```diff
@@ -17,7 +18,7 @@ async def refresh_invite_code(sid: str):
     if pr.role != Role.DM:
         logger.warning(f"{pr.player.name} attempted to refresh the invitation code.")
         return
-
+    #
```
[the remark this answers is not in the document]
Comment (mischievous) [reply server/api/socket/room.py:21]: Removed

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,12 @@ These usually have no immediately visible impact on regular users
 
 ## Unreleased
 
+### Added
+-   Code to set a public hostname to be used when updating the invitation url by reading from server_config.cfg 
+    "general -> publicName".  If the publicName is empty or does not exists it falls back to normal operation.
+-   Added code to planarally.py to display the warning about the template directory if not running in 
+    dev mode.
+
```
[the remark this answers is not in the document]
Comment (mischievous) [reply CHANGELOG.md:16]: Updated

File: server/api/socket/room.py [hunk 4]
```diff
@@ -29,6 +30,25 @@ async def refresh_invite_code(sid: str):
...
+async def refresh_public_name (sid: str):
+    pr: PlayerRoom = game_state.get(sid)
+
+    if pr.role != Role.DM:
+        logger.warning(f"{pr.player.name} attempted to refresh the public name.")
+        return
+
+    #
+    publicName = config.get("General", "publicName", fallback = "")
+    if len (publicName) == 0:
```
[the remark this answers is not in the document]
Comment (mischievous) [reply server/api/socket/room.py:44]: Removed white space.

File: server/api/socket/room.py [hunk 5]
```diff
@@ -29,6 +30,25 @@ async def refresh_invite_code(sid: str):
...
+        logger.warning("refresh_invite_code : publicName not set")
+
+    #
+    await sio.emit ( "Room.Info.PublicName.Set", str(publicName), room=sid, namespace=GAME_NS )
```
[the remark this answers is not in the document]
Comment (mischievous) [reply server/api/socket/room.py:48]: Removed white space.

File: server/server_config.cfg [hunk 6]
```diff
@@ -20,6 +20,7 @@ ssl_privkey = cert/privkey.pem
 
 [General]
 save_file = planar.sqlite
+publicName =
```
[the remark this answers is not in the document]
Comment (mischievous) [reply server/server_config.cfg:23]: Done

File: CHANGELOG.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Kruptein) [reply CHANGELOG.md:16]: This one seems to still be in a duplicate section

File: server/api/socket/room.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kruptein) [reply server/api/socket/room.py:20]: This one is still there it sems

[the remark this answers is not in the document]
Comment (mischievous) [reply server/api/socket/room.py:20]: Gone now.

**Pull request merged**